### ISSUE TITLE
CI: Intermittent hang in libgit2 challenge prompt test.

### ISSUE THREAD

**Issue by Keno** [JuliaLang/julia#45797]
E.g.: https://buildkite.com/julialang/julia-master/builds/13170#01819131-f9a7-48e4-926f-23e48419c663

We have an rr trace.
```


LibGit2/libgit2                           (9) \|         failed at 2022-06-23T17:20:31.754
--
  | Error During Test at /cache/build/default-amdci4-5/julialang/julia-master/julia-a375fe8755/share/julia/stdlib/v1.9/LibGit2/test/libgit2.jl:2545
  | Got exception outside of a @test
  | Process timed out possibly waiting for a response. Process output found:
  | """
  | Username for 'https://github.com': foo
  | Password for 'https://foo@github.com':
  | signal (15): Terminated
  | in expression starting at none:1
  | unknown function (ip: 0x7000000c)
  | unknown function (ip: 0x7ff02ddd2733)
  | untraced_syscall_base at /workspace/srcdir/rr/src/preload/syscallbuf.c:369 [inlined]
  | sys_read at /workspace/srcdir/rr/src/preload/syscallbuf.c:2436
  | syscall_hook_internal at /workspace/srcdir/rr/src/preload/syscallbuf.c:3387 [inlined]
  | syscall_hook at /workspace/srcdir/rr/src/preload/syscallbuf.c:3494
  | unknown function (ip: 0x7ff02ddcd58f)
  | unknown function (ip: 0x7ff02ddcd5ee)
  | unknown function (ip: 0x7ff02ddcd60a)
  | __read at /lib/x86_64-linux-gnu/libpthread.so.0 (unknown line)
  | ijl_getch at /cache/build/default-amdci4-3/julialang/julia-master/src/sys.c:538
  | _getch at ./util.jl:260 [inlined]
  | getpass at ./util.jl:272
  | _jl_invoke at /cache/build/default-amdci4-3/julialang/julia-master/src/gf.c:2374 [inlined]
  | ijl_apply_generic at /cache/build/default-amdci4-3/julialang/julia-master/src/gf.c:2575
  | getpass at ./util.jl:284
  | authenticate_userpass at /cache/build/default-amdci4-3/julialang/julia-master/usr/share/julia/stdlib/v1.9/LibGit2/src/callbacks.jl:222
  | credentials_callback at /cache/build/default-amdci4-3/julialang/julia-master/usr/share/julia/stdlib/v1.9/LibGit2/src/callbacks.jl:331
  | jfptr_credentials_callback_56476.clone_1 at /cache/build/default-amdci4-5/julialang/julia-master/julia-a375fe8755/lib/julia/sys.so (unknown line)
  | _jl_invoke at /cache/build/default-amdci4-3/julialang/julia-master/src/gf.c:2374 [inlined]
  | ijl_apply_generic at /cache/build/default-amdci4-3/julialang/julia-master/src/gf.c:2575
  | unknown function (ip: 0x7ff00c882288)
  | #credential_loop#7 at /cache/build/default-amdci4-5/julialang/julia-master/julia-a375fe8755/share/julia/stdlib/v1.9/LibGit2/test/libgit2-helpers.jl:31
  | credential_loop##kw at /cache/build/default-amdci4-5/julialang/julia-master/julia-a375fe8755/share/julia/stdlib/v1.9/LibGit2/test/libgit2-helpers.jl:13 [inlined]
  | #credential_loop#8 at /cache/build/default-amdci4-5/julialang/julia-master/julia-a375fe8755/share/julia/stdlib/v1.9/LibGit2/test/libgit2-helpers.jl:68 [inlined]
  | credential_loop at /cache/build/default-amdci4-5/julialang/julia-master/julia-a375fe8755/share/julia/stdlib/v1.9/LibGit2/test/libgit2-helpers.jl:62 [inlined]
  | credential_loop at /cache/build/default-amdci4-5/julialang/julia-master/julia-a375fe8755/share/julia/stdlib/v1.9/LibGit2/test/libgit2-helpers.jl:62


```

**Comment by giordano** [comment]
This should have been fixed by #45954, right?

**Comment by Keno** [comment]
Yes
